gcp_variant_transforms/beam_io/vcf_header_io.py

```diff
@@ -203,58 +199,74 @@ def _get_samples_pysam(self, sample_line):
     else:
       return []
 
+  def verify_infos(self, fields):
+    for k, v in fields.iteritems():
+      # ID, Description, Type and Number are mandatory fields.
+      if not k:
+        raise ValueError('Corrupt ID at header line {}.'.format(v.id))
+      if v.description is None:
+        raise ValueError(
+            'Corrupt Description at header line {}.'.format(v.id))
+      # Type can only be Integer, Float, Flag, Character or String
+      if not v.type or v.record['Type'] not in HEADER_TYPES:
+        raise ValueError('Corrupt Type at header line {}'.format(v.id))
+      # Number can only be a number or one of 'A', 'R', 'G' and '.'.
```

Comment:
Shouldn't we verify `v.number` instead of `v.record['Number']`

Also as a more general question, here do we need to verify `record` at all?  My intuition is that we are verifying the mandatory fields which we expect to find in `pysam.libcbcf.VariantHeader` and there is no reason to verify `pysam.libcbcf.VariantHeaderRecord`. Please correct me if I am wrong.